### ISSUE TITLE
helm v3.12.0 is breaking helm install

### ISSUE THREAD

**Issue by LittleHoopoe** [helm/helm#12095]
The new Version is not allowing me to install my helm chart to our k8s-cluster (Rancher)
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: the server has asked for the client to provide credentials

Output of `helm version`: `version.BuildInfo{Version:"v3.12.0", GitCommit:"c9f554d75773799f72ceef38c51210f1842a1dea", GitTreeState:"clean", GoVersion:"go1.20.4"}`

Output of `kubectl version`: 
`Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.9", GitCommit:"a1a87a0a2bcd605820920c6b0e618a8ab7d117d4", GitTreeState:"clean", BuildDate:"2023-04-12T12:16:51Z", GoVersion:"go1.19.8", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.9", GitCommit:"9710807c82740b9799453677c977758becf0acbb", GitTreeState:"clean", BuildDate:"2022-12-08T10:08:06Z", GoVersion:"go1.18.9", Compiler:"gc", Platform:"linux/amd64"}`

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Rancher

ps. v.3.11.3 same helm chart and same cluster is working fine.

**Comment by mmadoo** [comment]
I have a similar issue in 3.12.0 so I rollback to 3.11

**Comment by LittleHoopoe** [comment]
I've rolled back to v3.11.3. For everyone who has the same issue. Uninstall your helm and run this: 

`
$ curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3
$ chmod 700 get_helm.sh
$ ./get_helm.sh --version v3.11.3
`

**Comment by timvandruenen** [comment]
I have exactly the same issue. Did a rollback to v3.11.3 as well.
Something we can do to help debugging this issue?

It only happens for `helm install`, all other commands seem to work.

**Comment by Aaron-ML** [comment]
Seeing this also, previous version is working fine.

**Comment by Aaron-ML** [comment]
This seems to be specific to rancher, for now as a workaround you can connect directly to the rancher-managed cluster and helm will work on 3.12. Something between helm and the rancher API is having issues.

```
# get context options
k config get-contexts
# set context to direct (likely control plane / etcd naming)
kubectl config use-context <directtocluster>
```

**Comment by joejulian** [comment]
Closing this as a duplicate of https://github.com/helm/helm/issues/12071. When a release with an updated client-go is released (v3.12.1), re-test this. If it's still a problem, reopen the issue.
